packages/headless/src/features/case-fields/case-fields-slice.ts

```diff
@@ -0,0 +1,51 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {
+  disableCaseClassifications,
+  enableCaseClassifications,
+  fetchCaseClassifications,
+  setCaseField,
+} from './case-fields-actions';
+import {getCaseFieldsInitialState} from './case-fields-state';
+
+export const caseFieldsReducer = createReducer(
+  getCaseFieldsInitialState(),
+
+  (builder) => {
+    builder
+      .addCase(setCaseField, (state, action) => {
```

Comment:
Hmm, I feel we could split this action into `registerCaseField` and `updateCaseField`, and remove the if/else